doc/dev/rev-store.md

```diff
@@ -0,0 +1,124 @@
+    automatically update to the newest revision
+  * It a revision is specified, the revision store will only update if the
+    revision is not yet cached, otherwise the cached version can be used
+
+The final consideration means that an offline usage is possible if all
+repositories specified are specified with their hash.
+
+Due to the fact that the revision store is a Git repository it means that the
+data sources that can be added to it also have to be available via Git. This
+means that adding repositories that use different version control systems
+aren't supported at the moment nor are plain HTTP sources supported.
+
+Support for other kinds of VCSes is a possible extension by replicating similar
+concepts with other version control systems, provided they allow for similar
+flexibility as the Git way of storing revisions. However at the moment most
+users have settled on using Git, hence this version should be able to
+accomodate the needs for most users.
+
+Another compromise is that old repositories with long histories and large sizes
+have to be cloned before use, thus increasing the size of the initial download
+compared to the same metadata downloaded as a compressed tarball. Despite Git
+compressing objects, the history of the repositories to be added does increase
+the overhead.
+
+A solution to this could be shallow clones which only contain the latest
+revisions, however these have [shown to be
+problematic](https://blog.cocoapods.org/Master-Spec-Repo-Rate-Limiting-Post-Mortem/)
+thus for time being we are fetching the complete histories.
+
+Implementation
+--------------
+
+This section describes the current way the revision store is implemented.
+
+As the revision store is not project specific, it is stored in the user's cache
+directory (using the [freedesktop.org](https://www.freedesktop.org/wiki/)
+specifications, the directory specified by `XDG_CACHE_HOME`), with all dune
+instances sharing one single revision store.
+
+The revision store itself is a `bare` Git repository without a worktree. This
+is because all repositories in the revision store are equal and checking out
+one particular revision would be a waste of disk space as the Git tooling can
+be used to construct any revisions out of the bare repository anyway.
+
+Thus every source that is added to the revision store as a remote that tracks
+the default branch (or, if a branch is specified explicitely, then that
+branch) and fetched, thus storing the required revisions in the revision store.
+
+The implementation of these features is a mix of calling the `git` binary and
+implementing parts in OCaml. This means that the `git` binary is required on
+the system. Possible future improvements could be using
+[ocaml-git](https://github.com/mirage/ocaml-git) to avoid the dependency.
+
+Getting the Contents
+--------------------
+
+The `Rev_store` module contains modules/types that are reflecting specific
```

Comment:
I feel like details of this contents are going to change and are best documented in `Rev_store` itself.

Reply:
Good point, I'll remove this section. The `mli` should be clear enough and if not then it can be documented there.